src/axom/inlet/Field.hpp

```diff
@@ -87,9 +94,121 @@ class Field : public std::enable_shared_from_this<Field>
+  */
+  std::shared_ptr<Field> defaultValue(const char* value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the default value of this Field.
+   *
+   * Set the default value for the Field in the input deck.
+   *
+   * \param [in] value The default boolean value
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> defaultValue(bool value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the default value of this Field.
+   *
+   * Set the default value for the Field in the input deck.
+   *
+   * \param [in] value The default integer value
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> defaultValue(int value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the default value of this Field.
+   *
+   * Set the default value for the Field in the input deck.
+   *
+   * \param [in] value The default double value
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> defaultValue(double value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the range of this Field.
+   *
+   * Set the continuous range for the Field in the input deck.
+   *
+   * \param [in] startVal The start of the range
+   * 
+   * \param [in] endVal The end of the range
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> range(double startVal, double endVal);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the range of this Field.
+   *
+   * Set the continuous range for the Field in the input deck.
+   *
+   * \param [in] startVal The start of the range
+   * 
+   * \param [in] endVal The end of the range
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> range(int startVal, int endVal);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the range of this Field.
+   *
+   * Set the continuous range for the Field in the input deck.
+   *
+   * \param [in] set An integer vector containing the set of allowed values
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> validValues(std::vector<int> set);
```

Comment:
should this be a `std::set` instead of a ``std::vector``?

Reply:
I used a vector since I was converting set into an array to store it in sidre. Is there a way to store an std::set in a sidre View?

Reply:
You should be copying the values out of the given container and into sidre. I would worry that the vector/set gets destroyed then you are accessing invalid memory.

Reply:
That's what it's doing right now; I copied the vector contents into a sidre buffer.

Reply:
This isn't super critical -- using `std::set` would remove duplicates, but the behavior and usage would otherwise be pretty similar.